portfolio.py

```diff
@@ -2,21 +2,55 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import TRADING_INSTRUMENTS
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
-    def __init__(self, date):
+    def __init__(self, np_array=None, encoding=None):
         dict.__init__(self)
-        self.date = date
+        if np_array is not None:
+            if isinstance(np_array, np.ndarray):
+                self.decode_from_nparray(np_array, encoding)
+            else:
+                raise TypeError(
+                    'The input is %s, not a numpy array.' % type(np_array))
+
+    def decode_from_nparray(self, np_array, encoding):
+        """
+        Decode numpy array into portfolio.
+        """
+        if encoding is None:
+            raise KeyError('Invalid encoding method: %r' % encoding)
+        if len(np_array) != len(TRADING_INSTRUMENTS[encoding]):
+            raise ValueError(
+                'Input dimension does not match number of instruments.')
+        for index, instrument in TRADING_INSTRUMENTS[encoding]:
+            self[instrument] = np_array[index]
+
+    def encode_to_nparray(self, encoding):
+        """
+        Encode the positions into numpy type.
+        """
+        return np.array([
+            self.get(instrument, 0)
+            for instrument in TRADING_INSTRUMENTS[encoding]])
+
+    def is_tradable(self, instruments):
+        """
+        Judge whether the instruments in self are tradable or not.
+        """
+        for instrument in self:
+            if instrument not in instruments:
+                raise KeyError("%s is not tradable." % instrument)
```

Comment:
When we define a method with name `is_xxx`, we expect it to return a bool.
Now it doesn't return anything at all. I suggest to return a bool and let the caller decide how to use the return value.